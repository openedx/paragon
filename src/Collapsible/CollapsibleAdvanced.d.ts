import React from 'react';

export interface CollapsibleAdvancedProps {
    children?: React.ReactNode;
    className?: string;
    defaultOpen?: boolean;
    open?: boolean;
    onToggle?: () => void;
    onOpen?: () => void;
    onClose?: () => void;
    unmountOnExit?: boolean;
}

declare class CollapsibleAdvanced extends React.Component<CollapsibleAdvancedProps> {}

export default CollapsibleAdvanced;
