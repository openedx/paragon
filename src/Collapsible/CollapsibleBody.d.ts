import React from 'react';

export interface CollapsibleBodyProps {
    children?: React.ReactNode;
    tag?: string,
    transitionWrapper?: React.ReactElement,
}

declare const CollapsibleBody: React.FC<CollapsibleBodyProps>;

export default CollapsibleBody;
