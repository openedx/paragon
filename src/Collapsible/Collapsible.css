/**
 * Do not edit directly
 * Generated on Wed, 01 Mar 2023 00:27:19 GMT
 */

:root {
  --pgn-spacing-collapsible-card-spacer-basic-icon: .625rem;
  --pgn-spacing-collapsible-card-spacer-basic-x: .5rem;
  --pgn-spacing-collapsible-card-spacer-basic-y: .5rem;
  --pgn-spacing-collapsible-card-spacer-icon: 2.5rem;
  --pgn-spacing-collapsible-card-spacer-left-body: .75rem;
  --pgn-spacing-collapsible-card-spacer-x-base: .5rem;
  --pgn-spacing-collapsible-card-spacer-y-base: .5rem;
  --pgn-spacing-collapsible-card-spacer-x-lg: var(--pgn-spacing-card-spacer-x);
  --pgn-spacing-collapsible-card-spacer-y-lg: var(--pgn-spacing-card-spacer-y);
}
