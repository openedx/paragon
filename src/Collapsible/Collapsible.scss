@import "variables";

.collapsible-card {
  // stylelint-disable-next-line
  @extend .card;

  border-radius: 0;

  .collapsible-trigger {
    padding: $collapsible-card-spacer-y $collapsible-card-spacer-x;
    border-radius: $card-inner-border-radius;
    border-bottom: $card-border-width solid transparent;
    transition: border-color 100ms ease 150ms;
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: space-between;
    text-align: start;

    & > * {
      margin-bottom: 0;
      margin-top: 0;
    }

    &[aria-expanded="true"] {
      border-radius: $card-inner-border-radius $card-inner-border-radius 0 0;
      border-color: $card-border-color;
      transition: none;
      text-align: start;
    }
  }

  .collapsible-body {
    // stylelint-disable-next-line
    @extend .card-body;

    padding: $collapsible-card-spacer-y $collapsible-card-spacer-x;
    padding-left: $collapsible-card-body-spacer-left;
    text-align: start;

    & > *:last-child {
      margin-bottom: 0;
    }
  }

  .collapsible-icon {
    margin-inline-start: $collapsible-card-spacer-icon;
  }
}

.collapsible-card-lg {
  // stylelint-disable-next-line
  @extend .collapsible-card;

  .collapsible-trigger {
    padding: $collapsible-card-spacer-y-lg $collapsible-card-spacer-x-lg;
  }

  .collapsible-body {
    padding: $collapsible-card-spacer-y-lg $collapsible-card-spacer-x-lg;
  }
}

.collapsible-basic {
  .collapsible-trigger {
    display: flex;
    cursor: pointer;
    align-items: center;
    text-decoration: underline;
    color: theme-color("primary", "default");
    padding: $collapsible-basic-spacer-y $collapsible-basic-spacer-x;
  }

  .collapsible-body {
    padding: $collapsible-basic-spacer-y $collapsible-basic-spacer-x;
    text-align: start;

    &.collapsing {
      transition: height 300ms ease;
    }
  }

  .collapsible-icon {
    margin-left: $collapsible-basic-spacer-icon;
  }
}
