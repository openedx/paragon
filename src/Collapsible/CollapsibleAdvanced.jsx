import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

export const CollapsibleContext = React.createContext();

class CollapsibleAdvanced extends React.Component {
  static getDerivedStateFromProps(props) {
    if (props.open !== undefined) {
      return {
        // Since this method fires on both props and state changes, local updates
        // to the controlled value will be ignored, because the props version
        // always overrides it. In this case, this is exactly what we want.
        isOpen: props.open,
      };
    }
    return null;
  }

  constructor(props) {
    super(props);

    this.state = {
      isOpen: props.open !== undefined ? props.open : props.defaultOpen,
    };
  }

  open = () => {
    this.setState({ isOpen: true });

    if (this.props.onOpen) {
      this.props.onOpen();
    }
    if (this.props.onToggle) {
      this.props.onToggle(true);
    }
  }

  close = () => {
    this.setState({ isOpen: false });

    if (this.props.onClose) {
      this.props.onClose();
    }
    if (this.props.onToggle) {
      this.props.onToggle(false);
    }
  }

  toggle = () => {
    if (this.state.isOpen) {
      this.close();
    } else {
      this.open();
    }
  }

  render() {
    const {
      children,
      className,
      ...props
    } = this.props;

    // Unneeded for passthrough props
    delete props.defaultOpen;
    delete props.onToggle;
    delete props.onOpen;
    delete props.onClose;

    return (
      <div
        {...props}
        className={classNames('pgn_collapsible', className, {
          'is-open': this.state.isOpen,
        })}
      >
        <CollapsibleContext.Provider
          value={{
            isOpen: this.state.isOpen,
            open: this.open,
            close: this.close,
            toggle: this.toggle,
          }}
        >
          {children}
        </CollapsibleContext.Provider>
      </div>
    );
  }
}

CollapsibleAdvanced.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  /** Specifies whether `Collapsible` should be initially open. */
  defaultOpen: PropTypes.bool,
  /** Specifies whether `Collapsible` is open. */
  open: PropTypes.bool,
  /** Callback fired when `Collapsible's` state is toggled. */
  onToggle: PropTypes.func,
  /** Callback fired when `Collapsible` opens. */
  onOpen: PropTypes.func,
  /** Callback fired when `Collapsible` closes. */
  onClose: PropTypes.func,
};

CollapsibleAdvanced.defaultProps = {
  children: undefined,
  className: undefined,
  defaultOpen: false,
  open: undefined,
  onToggle: undefined,
  onOpen: undefined,
  onClose: undefined,
};

export default CollapsibleAdvanced;
