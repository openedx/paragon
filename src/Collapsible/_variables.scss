// Collapsible

$collapsible-card-spacer-y:               .5rem !default;
$collapsible-card-spacer-x:               .5rem !default;
$collapsible-card-spacer-y-lg:         $card-spacer-y !default;
$collapsible-card-spacer-x-lg:         $card-spacer-x !default;
$collapsible-card-body-spacer-left:        .75rem !default;
$collapsible-card-spacer-icon:             2.5rem !default;

$collapsible-basic-spacer-y:               .5rem !default;
$collapsible-basic-spacer-x:               .5rem !default;
$collapsible-basic-spacer-icon:            .625rem !default;
