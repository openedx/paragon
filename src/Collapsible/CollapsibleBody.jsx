import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import Collapse from '../Collapse';
import { CollapsibleContext } from './CollapsibleAdvanced';
import TransitionReplace from '../TransitionReplace';

function CollapsibleBody({
  children, transitionWrapper, tag, ...props
}) {
  const { isOpen, unmountOnExit } = useContext(CollapsibleContext);

  // Keys are added to these elements so that TransitionReplace
  // will recognize them as unique components and perform the
  // transition properly.
  const content = React.createElement(tag, { key: 'body', ...props }, children);
  const transitionBody = isOpen ? content : <div key="empty" />;

  if (transitionWrapper) {
    return React.cloneElement(transitionWrapper, {}, transitionBody);
  }
  /* istanbul ignore next */
  return unmountOnExit
    ? <TransitionReplace>{transitionBody}</TransitionReplace>
    : <Collapse in={isOpen}>{content}</Collapse>;
}

CollapsibleBody.propTypes = {
  /** Specifies contents of the component. */
  children: PropTypes.node,
  /** Specifies content's base element. */
  tag: PropTypes.string,
  /** Specifies transition element. */
  transitionWrapper: PropTypes.element,
};

CollapsibleBody.defaultProps = {
  children: undefined,
  tag: 'div',
  transitionWrapper: undefined,
};

export default CollapsibleBody;
