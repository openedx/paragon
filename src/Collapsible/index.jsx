import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import { ExpandLess, ExpandMore } from '../../icons';
import CollapsibleAdvanced, { CollapsibleContext } from './CollapsibleAdvanced';
import CollapsibleBody from './CollapsibleBody';
import CollapsibleTrigger from './CollapsibleTrigger';
import CollapsibleVisible from './CollapsibleVisible';
import Icon from '../Icon';

const styleIcons = {
  basic: {
    iconWhenClosed: <Icon src={ExpandMore} />,
    iconWhenOpen: <Icon src={ExpandLess} />,
  },
  // card and card-lg use the defaults specified in defaultProps
};

const Collapsible = React.forwardRef((props, ref) => {
  const {
    children,
    className,
    title,
    styling,
    iconWhenClosed,
    iconWhenOpen,
    ...other
  } = props;

  const icons = { iconWhenClosed, iconWhenOpen, ...styleIcons[styling] };
  const titleElement = React.isValidElement(title) ? title : <span>{title}</span>;

  return (
    <Collapsible.Advanced
      {...other}
      className={classNames(className, `collapsible-${styling}`)}
      ref={ref}
    >
      <Collapsible.Trigger className="collapsible-trigger">
        {titleElement}
        <span className="collapsible-icon">
          <Collapsible.Visible whenClosed>{icons.iconWhenClosed}</Collapsible.Visible>
          <Collapsible.Visible whenOpen>{icons.iconWhenOpen}</Collapsible.Visible>
        </span>
      </Collapsible.Trigger>

      <Collapsible.Body className="collapsible-body">{children}</Collapsible.Body>
    </Collapsible.Advanced>
  );
});

Collapsible.propTypes = {
  /** Specifies contents of the component. */
  children: PropTypes.node.isRequired,
  /** Specifies class name to append to the base element. */
  className: PropTypes.string,
  /** Specifies whether the `Collapsible` should be initially open. */
  defaultOpen: PropTypes.bool,
  /** Specifies icon to show when `Collapsible` is closed. */
  iconWhenClosed: PropTypes.element,
  /** Specifies icon to show when `Collapsible` is open. */
  iconWhenOpen: PropTypes.element,
  /** Callback fired when `Collapsible` closes. */
  onClose: PropTypes.func,
  /** Callback fired when `Collapsible` opens. */
  onOpen: PropTypes.func,
  /** Callback fired when `Collapsible's` state is toggled. */
  onToggle: PropTypes.func,
  /** Specifies whether `Collapsible` is open. */
  open: PropTypes.bool,
  /** Specifies style variant. */
  styling: PropTypes.oneOf(['basic', 'card', 'card-lg']),
  /** Specifies title. */
  title: PropTypes.node.isRequired,
  /** Unmount the component (remove it from the DOM) when it is collapsed */
  unmountOnExit: PropTypes.bool,
};

Collapsible.defaultProps = {
  className: undefined,
  defaultOpen: false,
  iconWhenClosed: <Icon src={ExpandMore} />,
  iconWhenOpen: <Icon src={ExpandLess} />,
  onClose: undefined,
  onOpen: undefined,
  onToggle: undefined,
  open: undefined,
  styling: 'card',
  unmountOnExit: true,
};

Collapsible.Advanced = CollapsibleAdvanced;
Collapsible.Body = CollapsibleBody;
Collapsible.Trigger = CollapsibleTrigger;
Collapsible.Visible = CollapsibleVisible;
Collapsible.Context = CollapsibleContext;

export default Collapsible;
