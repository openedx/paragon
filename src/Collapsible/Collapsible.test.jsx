import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import Collapsible from '.';

const EXAMPLE_CONTENT = 'Example content';

const collapsibleContent = (
  <>
    <Collapsible.Trigger className="trigger d-flex align-items-center">
      <h4 className="flex-grow-1">A heading</h4>

      <span className="collapsible-card-header-icon" aria-hidden>
        <Collapsible.Visible whenClosed>+</Collapsible.Visible>
        <Collapsible.Visible whenOpen>-</Collapsible.Visible>
      </span>
    </Collapsible.Trigger>

    <Collapsible.Trigger data-testid="close-only" className="close-only" closeOnly>Close</Collapsible.Trigger>
    <Collapsible.Trigger data-testid="open-only" className="open-only" openOnly>Open</Collapsible.Trigger>

    <Collapsible.Body transitionWrapper={<div />}>
      <p className="example-content">
        {EXAMPLE_CONTENT}
      </p>
    </Collapsible.Body>
  </>
);

describe('<Collapsible />', () => {
  describe('Uncontrolled Rendering', () => {
    it('renders closed by default', () => {
      const tree = renderer.create((
        <Collapsible.Advanced>
          {collapsibleContent}
        </Collapsible.Advanced>
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders open by default', () => {
      const tree = renderer.create((
        <Collapsible.Advanced defaultOpen>
          {collapsibleContent}
        </Collapsible.Advanced>
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Controlled Rendering', () => {
    it('renders closed by default', () => {
      const tree = renderer.create((
        <Collapsible.Advanced open={false}>
          {collapsibleContent}
        </Collapsible.Advanced>
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders open by default', () => {
      const tree = renderer.create((
        <Collapsible.Advanced>
          {collapsibleContent}
        </Collapsible.Advanced>
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Imperative Methods', () => {
    const ref = React.createRef();
    beforeEach(() => {
      render(
        <Collapsible.Advanced ref={ref}>{collapsibleContent}</Collapsible.Advanced>,
      );
    });
    it('opens on .open()', () => {
      expect(screen.queryByText(EXAMPLE_CONTENT)).not.toBeInTheDocument();
      ref.current.open();
      expect(screen.getByText(EXAMPLE_CONTENT)).toBeInTheDocument();
    });

    it('closes on .close()', () => {
      ref.current.open();
      expect(screen.getByText(EXAMPLE_CONTENT)).toBeInTheDocument();
      ref.current.close();
      expect(screen.queryByText(EXAMPLE_CONTENT)).not.toBeInTheDocument();
    });

    it('correct behavior with unmountOnExit', () => {
      let i = 0;
      function Comp() {
        i += 1;
        return <h1>Hello world</h1>;
      }
      render(
        <Collapsible.Advanced ref={ref} unmountOnExit={false}>
          <Collapsible.Body>
            <Comp />
          </Collapsible.Body>
        </Collapsible.Advanced>,
      );
      ref.current.open();
      expect(i).toEqual(1);
      ref.current.close();
      ref.current.open();
      expect(i).toEqual(1);
    });
  });

  describe('Mouse Interactions', () => {
    let collapsible;
    const ref = React.createRef();
    beforeEach(() => {
      render(
        <Collapsible.Advanced ref={ref}>{collapsibleContent}</Collapsible.Advanced>,
      );
      collapsible = ref.current;
    });

    it('opens on trigger click', async () => {
      expect(screen.getAllByRole('button')[0]).toBeInTheDocument();
      await userEvent.click(screen.getAllByRole('button')[0]); // Open
      expect(screen.getByText(EXAMPLE_CONTENT)).toBeInTheDocument();
    });

    it('closes on trigger click', async () => {
      collapsible.open();
      expect(screen.getByText(EXAMPLE_CONTENT)).toBeInTheDocument();
      await userEvent.click(screen.getAllByRole('button')[0]); // Close
      expect(screen.queryByText(EXAMPLE_CONTENT)).not.toBeInTheDocument();
    });

    it('does not open on close only trigger click', async () => {
      collapsible.close();
      await userEvent.click(screen.getByTestId('close-only')); // No-op
      expect(screen.queryByText(EXAMPLE_CONTENT)).not.toBeInTheDocument();
    });

    it('closes on close only trigger click', async () => {
      collapsible.open();
      await userEvent.click(screen.getByTestId('close-only')); // Close
      expect(screen.queryByText(EXAMPLE_CONTENT)).not.toBeInTheDocument();
    });

    it('does not close on open only trigger click', async () => {
      collapsible.open();
      await userEvent.click(screen.getByTestId('open-only')); // No-op
      expect(screen.getByText(EXAMPLE_CONTENT)).toBeInTheDocument();
    });

    it('opens on open only trigger click', async () => {
      collapsible.close();
      await userEvent.click(screen.getByTestId('open-only')); // Open
      expect(screen.getByText(EXAMPLE_CONTENT)).toBeInTheDocument();
    });
  });

  describe('Keyboard Interactions', () => {
    let collapsible;
    const ref = React.createRef();
    beforeEach(() => {
      render(
        <Collapsible.Advanced ref={ref}>{collapsibleContent}</Collapsible.Advanced>,
      );
      collapsible = ref.current;
    });

    it('opens on trigger enter keydown', async () => {
      screen.getAllByRole('button')[0].focus();
      await userEvent.keyboard('{enter}'); // Open
      expect(screen.getByText(EXAMPLE_CONTENT)).toBeInTheDocument();
    });

    it('closes on trigger enter keydown', async () => {
      collapsible.open();
      screen.getAllByRole('button')[0].focus();
      await userEvent.keyboard('{enter}'); // Close
      expect(screen.queryByText(EXAMPLE_CONTENT)).not.toBeInTheDocument();
    });

    it('does not open on close only trigger enter keydown', async () => {
      collapsible.close();
      screen.getByTestId('close-only').focus();
      await userEvent.keyboard('{enter}'); // No-op
      expect(screen.queryByText(EXAMPLE_CONTENT)).not.toBeInTheDocument();
    });

    it('closes on close only trigger enter keydown', async () => {
      collapsible.open();
      screen.getByTestId('close-only').focus();
      await userEvent.keyboard('{enter}'); // Close
      expect(screen.queryByText(EXAMPLE_CONTENT)).not.toBeInTheDocument();
    });

    it('does not close on open only trigger enter keydown', async () => {
      collapsible.open();
      screen.getByTestId('open-only').focus();
      await userEvent.keyboard('{enter}'); // No-op
      expect(screen.getByText(EXAMPLE_CONTENT)).toBeInTheDocument();
    });

    it('opens on open only trigger enter keydown', async () => {
      collapsible.close();
      screen.getByTestId('open-only').focus();
      await userEvent.keyboard('{enter}'); // Open
      expect(screen.getByText(EXAMPLE_CONTENT)).toBeInTheDocument();
    });
  });
});
