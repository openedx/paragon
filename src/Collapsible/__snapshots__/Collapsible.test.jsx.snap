// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Collapsible /> Controlled Rendering renders closed by default 1`] = `
<div
  className="pgn_collapsible"
  open={false}
>
  <div
    aria-expanded={false}
    className="trigger d-flex align-items-center"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <h4
      className="flex-grow-1"
    >
      A heading
    </h4>
    <span
      aria-hidden={true}
      className="collapsible-card-header-icon"
    >
      +
    </span>
  </div>
  <div
    aria-expanded={false}
    className="close-only"
    data-testid="close-only"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    Close
  </div>
  <div
    aria-expanded={false}
    className="open-only"
    data-testid="open-only"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    Open
  </div>
  <div>
    <div />
  </div>
</div>
`;

exports[`<Collapsible /> Controlled Rendering renders open by default 1`] = `
<div
  className="pgn_collapsible"
>
  <div
    aria-expanded={false}
    className="trigger d-flex align-items-center"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <h4
      className="flex-grow-1"
    >
      A heading
    </h4>
    <span
      aria-hidden={true}
      className="collapsible-card-header-icon"
    >
      +
    </span>
  </div>
  <div
    aria-expanded={false}
    className="close-only"
    data-testid="close-only"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    Close
  </div>
  <div
    aria-expanded={false}
    className="open-only"
    data-testid="open-only"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    Open
  </div>
  <div>
    <div />
  </div>
</div>
`;

exports[`<Collapsible /> Uncontrolled Rendering renders closed by default 1`] = `
<div
  className="pgn_collapsible"
>
  <div
    aria-expanded={false}
    className="trigger d-flex align-items-center"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <h4
      className="flex-grow-1"
    >
      A heading
    </h4>
    <span
      aria-hidden={true}
      className="collapsible-card-header-icon"
    >
      +
    </span>
  </div>
  <div
    aria-expanded={false}
    className="close-only"
    data-testid="close-only"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    Close
  </div>
  <div
    aria-expanded={false}
    className="open-only"
    data-testid="open-only"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    Open
  </div>
  <div>
    <div />
  </div>
</div>
`;

exports[`<Collapsible /> Uncontrolled Rendering renders open by default 1`] = `
<div
  className="pgn_collapsible is-open"
>
  <div
    aria-expanded={true}
    className="trigger d-flex align-items-center"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <h4
      className="flex-grow-1"
    >
      A heading
    </h4>
    <span
      aria-hidden={true}
      className="collapsible-card-header-icon"
    >
      -
    </span>
  </div>
  <div
    aria-expanded={true}
    className="close-only"
    data-testid="close-only"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    Close
  </div>
  <div
    aria-expanded={true}
    className="open-only"
    data-testid="open-only"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    Open
  </div>
  <div>
    <div>
      <p
        className="example-content"
      >
        Example content
      </p>
    </div>
  </div>
</div>
`;
